src/main/ngapp/accio/src/app/mask.directive.ts

```diff
@@ -0,0 +1,110 @@
+import { Directive, HostListener, ElementRef, Input } from '@angular/core';
+import { MagicWandService } from './magic-wand.service';
+
+@Directive({
+  selector: '[appMask]',
+})
+export class MaskDirective {
+  @Input() scale: number;
+  @Input() originalData: ImageData;
+  @Input() scaledData: ImageData;
+  @Input() maskData: ImageData;
+
+  // create direct reference of canvas on editor.html
+  constructor(
+    private canvas: ElementRef<HTMLCanvasElement>, 
+    private magicWandService: MagicWandService,
+  ) { }
+  private ctx: CanvasRenderingContext2D;
+  private maskImage: HTMLImageElement;
+
+  @HostListener('click', ['$event'])
+  onClick(e: MouseEvent) {
+    const xCoord = e.offsetX;
+    const yCoord = e.offsetY;
+    this.ctx = this.canvas.nativeElement.getContext('2d');
+
+    // TODO(shmcaffrey): Tolerance will come from user input.
+    const tolerance = 20;
+
+    //  TODO-MASK-maskPixels needs to contain cumulative set of all pixels user clicked and added to the mask.
+    // returns an array indices of each pixel in the mask.
+    const maskPixels = this.magicWandService.floodfill(this.originalData, Math.floor(xCoord / this.scale), Math.floor(yCoord / this.scale), tolerance);
+
+    // TODO(shmcaffrey): change Alpha value to incorperate user input.
+    this.drawMask(maskPixels, this.scaledData, 1);
```

Comment:
Is there a reason you are passing in the scaledData instead of the originalData and having canvas scale when drawing?  Is is for performance?

Reply:
Noted in chat